syntax = "proto3";

import 'proto/user.proto';
import 'proto/profile.proto';
import 'proto/record.proto';

enum InterType {
  INTER_TYPE_INVALID = 0;             // 非法互动类型
  INTER_TYPE_LIKE = 1;                // 赞
  INTER_TYPE_COMMENT = 2;             // 评论
}

// 一条互动记录的原始信息
message RoughInteract {
  string id = 1;                      // 互动id
  string user = 2;                    // 用户
  int64 time = 3;                     // 互动时间
  InterType type = 4;                 // 互动类型
  string text = 5;                    // 互动内容,只对评论有效
  string record_id = 6;               // 互动记录id
  string target_interact_id = 7;      // 互动对象互动id
  bool is_delete = 8;                 // 是否删除
}

// 精减互动数据
message StripInteract {
  string id = 1;                      // 互动id
  InterType type = 2;                 // 互动类型
  int64 time = 3;                     // 互动时间 
  string text = 4;                    // 评论内容
  string target_nickname = 5;         // 回复目标用户的昵称
}

// 扩展互动数据
message ExtInteract {
  StripInteract interact = 1;        // 互动
  StripUserProfile user = 2;         // 用户
  StripRecord record = 3;            // 记录
}


// 互动远程服务
//service Interact {
//  rpc HandleInteract (InteractReq) returns (InteractRes) {}
//}
//
//enum InterOP {
//  INTER_OP_INVALID = 0;               // 非法操作
//  INTER_OP_ADD = 1;                   // 新增
//  INTER_OP_DELETE = 2;                // 删除
//}
//
//message InteractReq {
//  InterOP op = 1;
//  string id = 2;
//  int64 time = 3;
//}
//
//message InteractRes {
//  int32 code = 1;
//  string id = 2;
//}
//
